workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/CodeGeneration/Model/SchemaBundleV1/Common.cs

```diff
@@ -1,13 +1,73 @@
+using System;
 using System.Collections.Generic;
+using System.Linq;
 using Newtonsoft.Json;
 
 namespace Improbable.Gdk.CodeGeneration.Model.SchemaBundleV1
 {
-    public class Identifier
+    public class Identifier : IEquatable<Identifier>
     {
         [JsonProperty("qualifiedName")] public string QualifiedName;
         [JsonProperty("name")] public string Name;
         [JsonProperty("path")] public List<string> Path;
+
+        public string PackagePath => string.Join(".", Path.Take(Path.Count - 1));
+
+        public static Identifier FromQualifiedName(string qualifiedName)
+        {
+            return new Identifier
+            {
+                QualifiedName = qualifiedName,
+                Name = qualifiedName.Split(".").Last(),
+                Path = qualifiedName.Split(".").ToList()
+            };
+        }
+
+        public bool IsChild(Identifier identifier)
```

Comment:
It feels weird to have `IsChild` and `FromQualifiedName` in something that (IMO) should be a very pure data container. Can you refactor so that hierarchy is decided from outside of this class?

Reply:
The `IsChild` method can be moved somewhere else easily.

The `FromQualifiedName` is required as type references in commands and events are only given a FQN, not a fully identifier struct. So when generating event/command code, I need to access the actual type --> create an identifier.

I can move this out of the class definition into something else but will end up with a `IdentifierFactory.FromQualifiedName(string fqn)` or similar.

---

The alternative is to key all the types on the qualified name instead, but sort of defeats the purpose of an Identifier. On reflection - this should probably be an FR.

For reference the JSON of the event definition: 

```
{
      "identifier": {
       "qualifiedName": "playground.CubeColor.change_color",
       "name": "change_color",
       "path": [
        "playground",
        "CubeColor",
        "change_color"
       ]
      },
      "eventIndex": 1,
      "type": {
       "type": {
        "qualifiedName": "playground.ColorData"
       }
      },
```

Reply:
Moved `IsChild` to the calling site and `FromQualifiedName` to `CommonDetailsUtils.CreateIdentifier`